### PR TITLE
Add go-micro plugins

### PR DESCRIPTION
**Pull request by zaunist**
The test has not been completed yet, there may be some problems, I will correct them later

### CODE REVIEW

**Comment by zaunist** [comment]
image

**Comment by zaunist** [comment]
image

File: micro/go.mod [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package micro (sw_micro) is a plugin that can be used to trace Go-micro framework.
+
+module swmicro
```
Comment (arugal) [inline micro/go.mod:20]: I think the name of the module should be `github.com/SkyAPM/go2sky-plugins/go-micro`, what do you think?

File: micro/swmicro.go [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package micro (sw_micro) is a plugin that can be used to trace Go-micro framework.
+
+package swmicro
```
Comment (arugal) [inline micro/swmicro.go:20]: Should be `go-micro`.

**Review by arugal — COMMENTED** [review]
👍 Almost finished, you need to add an example and documentation.

File: micro/go.mod (hunk 1 above)
Comment (zaunist) [reply micro/go.mod:20]: I think so

File: micro/swmicro.go (hunk 2 above)
Comment (zaunist) [reply micro/swmicro.go:20]: I will make changes on the next commit.

**Comment by arugal** [comment]
### screenshots

image
image
image

**Review by arugal — APPROVED** [review]
LTGM, Thank you for contribution :)

**Comment by arugal** [comment]
@zaunist Would you be interested in adding the `go-mirco` in the [component-library.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml#L419) :)

**Comment by zaunist** [comment]
> @zaunist Would you be interested in adding the `go-mirco` in the [component-library.yml](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml#L419) :)

I will add the go-micro in the component-library.yml later, thanks :)

**Pull request merged**